numpy/ma/core.py

```diff
@@ -6659,12 +6666,11 @@ def concatenate(arrays, axis=0):
         return data
     # OK, so we have to concatenate the masks
     dm = np.concatenate([getmaskarray(a) for a in arrays], axis)
+    dm = dm.reshape(d.shape)
```

Comment:
No idea why this is needed, but it was here before too, and changing that is out of scope for this fix...